WizardSchoolSignUpUIKit/WizardSchoolSignUpUIKit/ViewController.swift

```diff
@@ -45,32 +52,52 @@ class ViewController: UIViewController {
             validate(userName: userName.lowercased())
         } else if notificationObject == passwordTextField, let password = passwordTextField.text {
             validate(password: password)
+            createAccountButton.isEnabled = configureCreateUserButton()
+
         } else if notificationObject == passwordConfirmationTextField, let passwordConfirmation = passwordConfirmationTextField.text {
             validate(password: passwordConfirmation, with: passwordTextField.text)
+            createAccountButton.isEnabled = configureCreateUserButton()
+
         }
     }
 
     private func validate(password: String) {
         let isValidPassword = password.count > 8
         passwordIcon.tintColor = isValidPassword ? .systemGreen : .systemRed
+        isPasswordValid = isValidPassword
     }
 
     private func validate(password: String, with userPassword: String?) {
         let isValidPassword = password == userPassword
         passwordConfirmationIcon.tintColor = isValidPassword ? .systemGreen : .systemRed
+        isPasswordConfirmationValid = isValidPassword
     }
 
     private func validate(userName: String) {
         network.validate(userName: userName) { result in
             switch result {
             case .success(let isAvailable):
                 DispatchQueue.main.async { [weak self] in
-                    self?.userName = userName
-                    self?.userNameIcon.tintColor = isAvailable ? .systemGreen : .systemRed
+                    guard let self = self else { return }
+                    self.userName = userName
+                    self.userNameIcon.tintColor = isAvailable ? .systemGreen : .systemRed
+                    self.isUserNameValid = isAvailable
+                    self.createAccountButton.isEnabled = self.configureCreateUserButton()
+
                 }
             case .failure:
                 break
             }
         }
     }
+    
+    private func configureCreateUserButton() -> Bool {
+        if isUserNameValid && isPasswordValid && isPasswordConfirmationValid {
+            createAccountButton.backgroundColor = UIColor.systemGreen
```

Comment:
I like this idea of configuring the element right in this method. What do you think about removing the return of this method and configure the isEnabled here as well?

```swift
let shouldEnableButton = isUserNameValid && isPasswordValid && isPasswordConfirmationValid
createAccountButton.backgroundColor = shouldEnableButton ? UIColor.systemGreen : UIColor.systemGreen.withAlphaComponent(0.5)
createAccountButton.isEnabled = shouldEnableButton
```
etc

Also, would be nice to make this call in the main thread to prevent running into issues if we call it form another thread in the ViewController

Reply:
The comments have been addressed in the most recent commit 5f2fc12